KarhooUISDK/Screens/CheckoutScreen/NewCheckout/NewCheckoutPresenter.swift

```diff
@@ -0,0 +1,94 @@
+    private let journeyDetails: JourneyDetails
+    private let quoteValidityWorker: QuoteValidityWorker
+    internal var passengerDetails: PassengerDetails!
+    private let threeDSecureProvider: ThreeDSecureProvider?
+    private let tripService: TripService
+    private let userService: UserService
+    private let bookingMetadata: [String: Any]?
+    private let paymentNonceProvider: PaymentNonceProvider
+    private let sdkConfiguration: KarhooUISDKConfiguration
+    private let analytics: Analytics
+    private let appStateNotifier: AppStateNotifierProtocol
+    private var trip: TripInfo?
+    private var comments: String?
+    private var bookingRequestInProgress: Bool = false
+    private var flightDetailsScreenIsPresented: Bool = false
+    private let baseFareDialogBuilder: PopupDialogScreenBuilder
+    private var cardRegistrationFlow: CardRegistrationFlow
+
+    // MARK: - Init & Config
+
+    init(
+        quote: Quote,
+        journeyDetails: JourneyDetails,
+        bookingMetadata: [String: Any]?,
+        quoteValidityWorker: QuoteValidityWorker = KarhooQuoteValidityWorker(),
+        threeDSecureProvider: ThreeDSecureProvider? = nil,
+        tripService: TripService = Karhoo.getTripService(),
+        userService: UserService = Karhoo.getUserService(),
+        passengerDetails: PassengerDetails? = PassengerInfo.shared.getDetails(),
+        analytics: Analytics = KarhooUISDKConfigurationProvider.configuration.analytics(),
+        appStateNotifier: AppStateNotifierProtocol = AppStateNotifier(),
+        baseFarePopupDialogBuilder: PopupDialogScreenBuilder = UISDKScreenRouting.default.popUpDialog(),
+        paymentNonceProvider: PaymentNonceProvider = PaymentFactory().nonceProvider(),
+        sdkConfiguration: KarhooUISDKConfiguration =  KarhooUISDKConfigurationProvider.configuration,
+        cardRegistrationFlow: CardRegistrationFlow = PaymentFactory().getCardFlow(),
+        callback: @escaping ScreenResultCallback<KarhooCheckoutResult>
+    ) {
+        self.threeDSecureProvider = threeDSecureProvider ?? sdkConfiguration.paymentManager.threeDSecureProvider
+        self.tripService = tripService
+        self.callback = callback
+        self.userService = userService
+        self.quoteValidityWorker = quoteValidityWorker
+        self.passengerDetails = passengerDetails ?? PassengerInfo.shared.currentUserAsPassenger()
+        self.paymentNonceProvider = paymentNonceProvider
+        self.sdkConfiguration = sdkConfiguration
+        self.appStateNotifier = appStateNotifier
+        self.analytics = analytics
+        self.baseFareDialogBuilder = baseFarePopupDialogBuilder
+        self.quote = quote
+        self.journeyDetails = journeyDetails
+        self.bookingMetadata = bookingMetadata
+        self.cardRegistrationFlow = cardRegistrationFlow
+    }
+
+    func viewDidLoad() {
+       quoteValidityWorker.setQuoteValidityDeadline(quote) {
+           // TODO: handle validity expiration
```

Comment:
Just to confirm, the logic for this will be handled in a new ticket, right?

Reply:
yeah, it suppose to be empty for now. The current code is designed only to make UI accessible, so @BartlomiejSopalaKarhoo can work on adding new UI components.